pilot/pkg/proxy/envoy/config.go

```diff
@@ -270,11 +270,22 @@ func buildSidecarListenersClusters(
 		// only HTTP outbound clusters are needed
 		httpOutbound := buildOutboundHTTPRoutes(mesh, node, instances, services, config)
 		httpOutbound = buildEgressHTTPRoutes(mesh, node, instances, config, httpOutbound)
-		clusters = append(clusters,
-			httpOutbound.clusters()...)
-		listeners = append(listeners,
-			buildHTTPListener(mesh, node, instances, nil, listenAddress, int(mesh.ProxyHttpPort),
-				RDSAll, useRemoteAddress, traceOperation, true, config))
+		clusters = append(clusters, httpOutbound.clusters()...)
+		opts := buildHTTPListenerOpts{
+			mesh:             mesh,
+			node:             node,
+			instances:        instances,
+			routeConfig:      nil,
+			ip:               listenAddress,
+			port:             int(mesh.ProxyHttpPort),
+			rds:              RDSAll,
+			useRemoteAddress: useRemoteAddress,
+			direction:        traceOperation,
+			outboundListener: true,
+			store:            config,
+		}
+
+		listeners = append(listeners, buildHTTPListener(opts))
```

Comment:
Ug, Github lost the review comment :(

I think this reads better in line:

```go
listeners = append(listeners, buildHTTPListener(buildHTTPListenerOpts{
    mesh:             mesh,
    node:             node,
    instances:        instances,
    routeConfig:      nil,
    ip:               listenAddress,
    port:             int(mesh.ProxyHttpPort),
    rds:              RDSAll,
    useRemoteAddress: useRemoteAddress,
    direction:        traceOperation,
    outboundListener: true,
    store:            config,
}))
```

or

```go
listeners = append(listeners,
    buildHTTPListener(buildHTTPListenerOpts{
        mesh:             mesh,
        node:             node,
        instances:        instances,
        routeConfig:      nil,
        ip:               listenAddress,
        port:             int(mesh.ProxyHttpPort),
        rds:              RDSAll,
        useRemoteAddress: useRemoteAddress,
        direction:        traceOperation,
        outboundListener: true,
        store:            config,
    }))
```

Not required to change, but curious for your opinion. The logic being that this pattern really is about providing named parameters at the function call site, so might as well just structure it like that.

Reply:
@ZackButcher I totally love the in-lining and didn't do it so we could get a feel for how ya'll like to do some of these things. I can change that around right now.

Reply:
ugh.. is this go-ism ? to have huge function call lines? :(.. :P

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#line-length

Reply:
@rshriram - yeah this is a bit of a smell but I really find the inlining reads a bit better and have seen it done mostly that way as @ZackButcher pointed out.

Reply:
@rshriram the root problem is that the call line is too long because we have so many parameters; the solution is to refactor our functions to have less parameters (e.g. by introducing some types to encapsulate config state).

It's out of scope of this PR to address the root problem, but this is an incremental improvement in that it makes it harder to call the function with bad/shuffled arguments.

Reply:
Yeah I am with you on that.. Just not a fan of inline struct declaration as part of function parameter. But I lack go hygiene. So, please take my complaint with a grain of salt.